services/PostService.js

```diff
@@ -280,15 +280,29 @@ async function getPostsByQuery(query, clickNo) {
   }
 }
 
-async function getPostCountByUserId(userId) {
+async function getPostCountByUserName(username) {
   let url = "";
   if (typeof window == "undefined") url = serverUrl;
   else url = baseUrl;
   try {
     const { data } = await axios.get(`${url}/${allPostsUrl}/count`, {
-      params: { userId: userId }
+      params: { username: username }
+    });
+    return data;
+  } catch (err) {
+    console.log(err);
+    return;
+  }
+}
+
+async function getAllPostsByUsername(username) {
+  let url = "";
+  if (typeof window == "undefined") url = serverUrl;
+  else url = baseUrl;
```

Comment:
Add a line break and write comments on what this code does.  better to have write it as a single function that can be reused

Reply:
made it as a function getUrl located at lib/getUrl